hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSourceSourceImpl.java

```diff
@@ -314,4 +314,15 @@ public String getMetricsName() {
   @Override public long getEditsFiltered() {
     return this.walEditsFilteredCounter.value();
   }
+
+  @Override
+  public void setWALReaderEditsBufferBytes(long usage) {
+    //noop. Global limit, tracked globally. Do not need per-source metrics
```

Comment:
nit: wouldn't it still be useful to know if particular sources were eating up the global limit? or do we not have enough information tracked already to do that?

Reply:
Once we chuck something into this usage, we have zero insight back to which source put it there. Wellington was working on this buffer tracking in HBASE-24813, but I don't think per-source tracking was "in scope"

Reply:
> Once we chuck something into this usage, we have zero insight back to which source put it there.

I had the same question. I was wondering if a drill down by source would be helpful in addition to the global usage. Correct me if I'm wrong, looks like ReplicationSource class has access to the MetricsSource object.. we can just update the byte usage for that source? (and the global too at the same time). That way we can also get rid of the special logic to update one metric setWALReaderEditsBufferBytes(). Does that not work for some reason?

Reply:
> looks like ReplicationSource class has access to the MetricsSource object.. we can just update the byte usage for that source? (and the global too at the same time). That way we can also get rid of the special logic to update one metric setWALReaderEditsBufferBytes()

You are correct that we could do that. I wanted to keep this change scoped on "make what we currently have reportable". I am all for doing a per-source tracking in addition to the globally-scoped tracking. I'd rather just keep these two things separate :)

Reply:
Oh, I see. Looks like you pushed the metrics update logic into the source, the patch looks clean now.

Reply:
yup! Just made another interface to isolate the new additions which cleans this up a little. I think your suggestion is still good for better, fine-grained tracking. However, since HBASE-20417, hopefully no one else runs into this ;)